src/i2c.rs

```diff
@@ -138,11 +147,26 @@ impl From<i2cdev::linux::LinuxI2CError> for I2CError {
 
 impl embedded_hal::i2c::Error for I2CError {
     fn kind(&self) -> embedded_hal::i2c::ErrorKind {
-        use embedded_hal::i2c::ErrorKind::*;
-        match &self.err {
-            // i2cdev::linux::LinuxI2CError::Nix(_) => todo!(),
-            // i2cdev::linux::LinuxI2CError::Io(_) => todo!(),
-            _ => Other,
+        use embedded_hal::i2c::ErrorKind;
+        use nix::errno::Errno::*;
+
+        let errno = match &self.err {
+            i2cdev::linux::LinuxI2CError::Nix(e) => *e,
+            i2cdev::linux::LinuxI2CError::Io(e) => {
+                match e.raw_os_error() {
+                    Some(r) => nix::Error::from_i32(r),
+                    None => return ErrorKind::Other,
+                }
+            },
+        };
+
+        // https://01.org/linuxgraphics/gfx-docs/drm/i2c/fault-codes.html
```

Comment:
This is probably the preferred link : https://www.kernel.org/doc/html/latest/i2c/fault-codes.html